src/status_im2/contexts/wallet/home/view.cljs

```diff
@@ -24,25 +25,52 @@
       :sub-label           (i18n/label :t/add-address-description)
       :add-divider?        true}]]])
 
-(def account-cards
-  [{:name                "Account 1"
-    :balance             "€0.00"
-    :percentage-value    "€0.00"
-    :customization-color :blue
-    :type                :empty
-    :emoji               "🍑"
-    :on-press            #(rf/dispatch [:navigate-to :wallet-accounts])}
-   {:customization-color :blue
-    :on-press            #(rf/dispatch
-                           [:show-bottom-sheet {:content new-account}])
-    :type                :add-account}])
+(def add-account-placeholder
+  {:customization-color :blue
+   :on-press            #(rf/dispatch [:show-bottom-sheet {:content new-account}])
+   :type                :add-account})
 
 (def tabs-data
   [{:id :assets :label (i18n/label :t/assets) :accessibility-label :assets-tab}
    {:id :collectibles :label (i18n/label :t/collectibles) :accessibility-label :collectibles-tab}
    {:id :activity :label (i18n/label :t/activity) :accessibility-label :activity-tab}])
 
-(defn view
+(defn calculate-raw-balance
+  [rawBalance decimals]
+  (/ (js/parseInt rawBalance) (Math/pow 10 (js/parseInt decimals))))
+
+(defn calculate-balance
+  [{:keys [address]}]
+  (let [tokens       (rf/sub [:wallet-2/tokens])
+        token        (get tokens (keyword (string/lower-case address)))
+        total-values (atom 0)]
+    (doseq [item token]
+      (let [total-value-per-token (atom 0)]
+        (doseq [balances (vals (:balancesPerChain item))]
+          (reset! total-value-per-token (+ (calculate-raw-balance (:rawBalance balances)
+                                                                  (:decimals item))
+                                           @total-value-per-token)))
+        (reset! total-values (+ (* @total-value-per-token
+                                   (get-in item [:marketValuesPerCurrency :USD :price]))
+                                @total-values))))
+    (.toFixed @total-values 2)))
+
+(defn refactor-data
```

Comment:
This function is suspiciously apt to be promoted to a layer-3 subscription. It has two inputs that are subscriptions, and the balance is calculated in a another function that also uses a subscription.

If you move the calculation to the subscription, it'll be cached automatically. Once you put it on the view, caching of the computation is gone.

I would suggest building/calculating as much as you can in the sub, and then in the end you can map and assoc only what's specific about the view, in this case the `:on-press` for example. Also consider extracting the `on-press` function to a let binding at mount time so that the same function is not rebuilt on every iteration.